File: search-service/src/main/java/com/gharana/search_service/dto/Hotel.java
package com.gharana.search_service.dto;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@AllArgsConstructor
@Getter
@Setter
public class Hotel {
    private String id;
    private String name;
    private String description;
    private String locationId;
    private String address;
    private String thumbnailUrl;
    private String rating;
    private List<String> amenities;
}

File: client/src/types/hotel.ts
import { Amenity } from "./amenity";

export interface Hotel {
  id: string;
  name: string;
  address: string;
  shortDescription: string;
  longDescription: string;
  thumbnailUrl: string;
  rating: string;
  amenities: Amenity[];
  nights: string;
  minAvgRatePerNight: string
};